core/finality-grandpa/src/lib.rs

```diff
@@ -481,6 +497,26 @@ pub fn run_grandpa<B, E, Block: BlockT<Hash=H256>, N, RA>(
 
 	register_finality_tracker_inherent_data_provider(client.clone(), &inherent_data_providers)?;
 
+	if let Some(telemetry_on_connect) = telemetry_on_connect {
+		let authorities = authority_set.clone();
+		let events = telemetry_on_connect.telemetry_connection_sinks
+			.for_each(move |_| {
+				telemetry!(CONSENSUS_INFO; "afg.authority_set";
+					 "authority_set_id" => ?authorities.set_id(),
+					 "authorities" => {
+						let curr = authorities.current_authorities();
+						let voters = curr.voters();
+						let authorities: Vec<String> =
+							voters.iter().map(|(id, _)| format!("{}", id)).collect();
+						format!("{:?}", authorities)
```

Comment:
Sounds like you want to use [this](https://docs.rs/itertools/0.8.0/itertools/fn.join.html) here or `serde_json::to_string`.